VirtualFinland.UsersAPI.Deployment/Program.cs

```diff
@@ -1,3 +1,80 @@
+using Pulumi;
+using VirtualFinland.UsersAPI.Deployment.Common;
+using VirtualFinland.UsersAPI.Deployment.Common.Models;
+using VirtualFinland.UsersAPI.Deployment.Features;
 
-return await Pulumi.Deployment.RunAsync<UsersApiStack>();
+return await Deployment.RunAsync(async () =>
+{
+    var config = new Config();
+    bool isProductionEnvironment = Environments.IsProductionEnvironment();
+    var organization = Deployment.Instance.OrganizationName;
+    var environment = Deployment.Instance.StackName;
+    var projectName = Deployment.Instance.ProjectName;
+
+    InputMap<string> tags = new()
+        {
+            {
+                "vfd:stack", Deployment.Instance.StackName
+            },
+            {
+                "vfd:project", Deployment.Instance.ProjectName
+            }
+        };
+
+    var stackSetup = new StackSetup()
+    {
+        Organization = organization,
+        ProjectName = projectName,
+        Environment = environment,
+        IsProductionEnvironment = isProductionEnvironment,
+        Tags = tags,
+    };
+
+    var isInitialDeployment = await stackSetup.IsInitialDeployment();
+
+    var cloudwatch = new CloudWatch(stackSetup);
+    var vpcSetup = new VpcSetup(stackSetup);
+    var database = new PostgresDatabase(config, stackSetup, vpcSetup, cloudwatch);
+    var dbConnectionStringSecret = new SecretsManager(stackSetup, "dbConnectionStringSecret", database.DatabaseConnectionString);
+    var dbAdminConnectionStringSecret = new SecretsManager(stackSetup, "dbAdminConnectionStringSecret", database.DatabaseAdminConnectionString);
+    var auditLogSubscriptionFunction = new AuditLogSubscription(config, stackSetup, database, cloudwatch);
+    var redisCache = new RedisElastiCache(stackSetup, vpcSetup);
+
+    var analyticsSqS = SqsQueue.CreateSqsQueueForAnalyticsCommand(stackSetup);
+
+    // The API
+    var usersApiFunction = new UsersApiLambdaFunction(config, stackSetup, vpcSetup, dbConnectionStringSecret, redisCache, cloudwatch, analyticsSqS, database);
+    usersApiFunction.SetupErrorAlerting(stackSetup);
+    var apiEndpoint = new LambdaFunctionUrl(stackSetup, usersApiFunction);
+
+    // Admin function for management tasks, scheduled events and triggers
+    var adminFunction = new AdminFunction(config, stackSetup, vpcSetup, dbAdminConnectionStringSecret, analyticsSqS, database);
+
+    // Admin function schedulers and triggers
+    adminFunction.CreateSchedulersAndTriggers(stackSetup, analyticsSqS);
+
+    if (isInitialDeployment)
+    {
+        // Initialize database
+        database.InvokeInitialDatabaseSetupFunction(stackSetup, adminFunction.LambdaFunction);
+    }
+    else
+    {
+        // Ensure database user 
+        database.InvokeDatabaseUserSetupFunction(stackSetup, adminFunction.LambdaFunction);
+        // Ensure database audit log triggers
+        database.InvokeInitialDatabaseAuditLogTriggersSetupFunction(stackSetup, adminFunction.LambdaFunction);
+    }
```

Comment:
not initial -> invoke **initial** database audit log triggers hämmings.

Jos nuo triggerit ajetaan aina joka tapauksessa (näin ymmärsin), niin onko heikohko nimeäminen?